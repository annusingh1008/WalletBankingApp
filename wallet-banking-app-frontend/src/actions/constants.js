export const authConstants = {
    LOGIN_REQUEST: 'LOGIN_REQUEST',
    LOGIN_SUCCESS: 'LOGIN_SUCCESS',
    LOGIN_FAILURE: 'LOGIN_FAILURE',
    LOGOUT_REQUEST: 'LOGOUT_REQUEST',
    LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
    LOGOUT_FAILURE: 'LOGOUT_FAILURE'
}

export const userConstants = {
    USER_REGISTER_REQUEST: 'USER_REGISTER_REQUEST',
    USER_REGISTER_SUCCESS: 'USER_REGISTER_SUCCESS',
    USER_REGISTER_FAILURE: 'USER_REGISTER_FAILURE',
}

export const amountTransferConstants = {
    AMOUNT_TRANSFER_REQUEST: 'AMOUNT_TRANSFER_REQUEST',
    AMOUNT_TRANSFER_SUCCESS: 'AMOUNT_TRANSFER_SUCCESS',
    AMOUNT_TRANSFER_FAILURE: 'AMOUNT_TRANSFER_FAILURE'
}

export const rechargeConstants = {
    RECHARGE_REQUEST: 'RECHARGE_REQUEST',
    RECHARGE_SUCCESS: 'RECHARGE_SUCCESS',
    RECHARGE_FAILURE: 'RECHARGE_FAILURE',
}

export const getUserDetailsConstants = {
    GET_USER_DETAILS_REQUEST: 'GET_USER_DETAILS_REQUEST',
    GET_USER_DETAILS_SUCCESS: 'GET_USER_DETAILS_SUCCESS',
    GET_USER_DETAILS_FAILURE: 'GET_USER_DETAILS_FAILURE'
}

export const transactionsConstants = {
    TRANSACTIONS_REQUEST: 'TRANSACTIONS_REQUEST',
    TRANSACTIONS_SUCCESS: 'TRANSACTIONS_SUCCESS',
    TRANSACTIONS_FAILURE: 'TRANSACTIONS_FAILURE',
    TOTAL_TRANSACTIONS_SUCCESS: 'TOTAL_TRANSACTIONS_SUCCESS',
}

export const cashbackConstants = {
    CASHBACK_REQUEST: 'CASHBACK_REQUEST',
    CASHBACK_SUCCESS: 'CASHBACK_SUCCESS',
    CASHBACK_FAILURE: 'CASHBACK_FAILURE',
    TOTAL_CASHBACKS_SUCCESS: 'TOTAL_CASHBACKS_SUCCESS'
}

export const getBalanceConstants = {
    GET_BALANCE_SUCCESS: 'GET_BALANCE_SUCCESS'
}