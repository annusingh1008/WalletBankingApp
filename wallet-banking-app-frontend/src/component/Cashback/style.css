.cashback-table {
    margin-left: 400px;
    width: 60%;
} 

thead tr:nth-child(1) th {
    background: white;
    position: sticky;
    top: 55px;
}

.no-cashbacks{
    margin-top: 200px;
    margin-left: 600px;
    font-size: 30px;
    font-weight: bold;
}

.button {
    border: 1px solid black;
    background-color: black;
    color: #ffffff;
    padding: 5px 10px;
  }
  
  .button:hover {
      border: 1px solid gray;
      background-color: gray;
      color: #ffffff;
      padding: 5px 10px;
    }
  
  .button:disabled,
  button[disabled]{
    border: 1px solid #999999;
    background-color: #cccccc;
    color: #666666;
  }
  
  .button:focus {     
      background-color: #cccccc;    
      color: black;
  }

  .all-btn{
    margin-left: 170px;
  }
  
