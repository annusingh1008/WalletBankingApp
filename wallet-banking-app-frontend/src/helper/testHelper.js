import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

const localStorageData = {
  email: "annu",
  password: "1234",
};

export const localStorageMock = (function () {
  return {
    getItem: function (key) {
      return localStorageData[key];
    },
    setItem: jest.fn(),
  };
})();

const mockStore = configureMockStore();

export const renderWithRedux = (ui, { store } = {}) => {
  return {
    ...render(<Provider store={mockStore(store)}>{ui}</Provider>),
    store,
  };
};
