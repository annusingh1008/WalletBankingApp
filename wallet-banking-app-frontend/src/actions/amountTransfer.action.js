import axios from "axios";
import { amountTransferConstants } from "./constants";

export const amountTransfer = (amountDetails) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/transfer/amount", {
      ...amountDetails,
    });

    if (res.status === 200) {
      if (res.data === "User does not exist") {
        alert("User does not exist! \nPlease check your email.");
      } else if (res.data === "Amount Transferred Successfully...!!") {
        alert("Transaction Successful");

        dispatch({
          type: amountTransferConstants.AMOUNT_TRANSFER_SUCCESS,
          payload: {
            amount:
              parseInt(amountDetails.currentAmount) -
              parseInt(amountDetails.amount),
            email: amountDetails.email,
          },
        });
      }
    }
  };
};
