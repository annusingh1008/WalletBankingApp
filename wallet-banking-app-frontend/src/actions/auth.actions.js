import axios from "axios";
import { authConstants } from "./constants";

export const login = (user) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/signin", {
      ...user,
    });

    if (res.status === 200) {
      if (res.data === "Signed in Successfully...!!") {
        localStorage.setItem("user", JSON.stringify(user));
        localStorage.setItem("email", user.email);

        dispatch({
          type: authConstants.LOGIN_SUCCESS,
          payload: user,
        });
      } else if (res.data === "Account does not exists with this email...!!") {
        alert("Account does not exists!! \nPlease check your email..");
      } else if (res.data === "Incorrect Password...!!") {
        alert("Please check your Password!!");
      }
    }
  };
};

export const getAuthDetails = (state) => state.auth;

export const signout = () => {
  return async (dispatch) => {
    dispatch({ type: authConstants.LOGOUT_REQUEST });
    localStorage.clear();
    dispatch({ type: authConstants.LOGOUT_SUCCESS });
  };
};
