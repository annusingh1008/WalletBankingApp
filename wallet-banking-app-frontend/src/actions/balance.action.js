import axios from "axios";
import { getBalanceConstants } from "./constants";

export const getUserBalance = (email) => {
  return async (dispatch) => {
    const res = await axios.get(`http://localhost:8085/getBalance/${email}`);

    dispatch({
      type: getBalanceConstants.GET_BALANCE_SUCCESS,
      payload: res.data,
    });
  };
};
