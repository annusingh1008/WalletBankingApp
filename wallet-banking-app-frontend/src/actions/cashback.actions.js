import axios from "axios";
import { cashbackConstants } from "./constants";

export const getAllCashBacks = (currentPage, email) => {
  return async (dispatch) => {
    const userEmail = localStorage.getItem("email");
    const res = await axios.get(
      `http://localhost:8085/getAllCashbacks/${userEmail}?pageNumber=${currentPage}&pageSize=${10}`
    );

    dispatch({
      type: cashbackConstants.CASHBACK_SUCCESS,
      payload: res.data,
    });
  };
};

export const getCashbacks = (state) => state.cashbacks;

export const getTotalCashbacks = (email) => {
  return async (dispatch) => {
    const res = await axios.get(
      `http://localhost:8085/getTotalCashbacks/${email}`
    );

    dispatch({
      type: cashbackConstants.TOTAL_CASHBACKS_SUCCESS,
      payload: res.data,
    });
  };
};
