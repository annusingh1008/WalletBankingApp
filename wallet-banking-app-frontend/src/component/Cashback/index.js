import React, { useEffect, useState } from "react";
import Layout from "../Layout";
import { Table, Button, Card, InputGroup } from "react-bootstrap";
import { useDispatch, useSelector } from "react-redux";
import {
  getAllCashBacks,
  getTotalCashbacks,
} from "../../actions/cashback.actions";
import "./style.css";
import { getCashbacks } from "../../actions/cashback.actions";

const Cashback = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getTotalCashbacks(email));
    dispatch(getAllCashBacks(0, email));
  }, []);

  const cashbacks = useSelector(getCashbacks);
  // const cashbacks = useSelector((state) => state.cashbacks);
  const cashbackList = cashbacks?.cashbackList;
  const email = localStorage.getItem("email");
  console.log(cashbackList);

  const length = cashbacks?.totalPages;
  const totalPages = length % 10 == 0 ? length / 10 : parseInt(length / 10) + 1;

  const [currentPage, setCurrentPage] = useState(0);

  const firstPage = () => {
    setCurrentPage(0);
    dispatch(getAllCashBacks(0, email));
  };

  const prevPage = () => {
    setCurrentPage(currentPage - 1);
    dispatch(getAllCashBacks(currentPage - 1), email);
  };

  const lastPage = () => {
    setCurrentPage(totalPages - 1);
    dispatch(getAllCashBacks(totalPages - 1), email);
  };

  const nextPage = () => {
    setCurrentPage(currentPage + 1);
    dispatch(getAllCashBacks(currentPage + 1), email);
  };

  const convertUTCDateToLocalDate = (date) => {
    var newDate = new Date(
      date.getTime() + date.getTimezoneOffset() * 60 * 1000
    );

    var offset = date.getTimezoneOffset() / 60;
    var hours = date.getHours();

    newDate.setHours(hours - offset);

    return newDate;
  };

  return (
    <Layout sidebar>
      {cashbackList?.length > 0 ? (
        <>
          <Table striped bordered hover className="cashback-table">
            <thead className="text-center">
              <tr>
                <th>#</th>
                <th>Cashback Earned</th>
                <th>Current Amount</th>
                <th>Date</th>
              </tr>
            </thead>
            <tbody className="text-center">
              {cashbackList.map((cashback, index) => (
                <tr key={index}>
                  <td>{index + 1}</td>
                  <td>{cashback.cashback_amount}</td>
                  <td>{cashback.current_amount}</td>
                  <td>
                    {convertUTCDateToLocalDate(
                      new Date(cashback.date)
                    ).toLocaleString()}
                  </td>
                </tr>
              ))}
            </tbody>
          </Table>
          <Card.Footer className="all-btn">
            <div style={{ marginLeft: "450px" }}>
              Showing Page {currentPage + 1} of {totalPages}
            </div>
            <div style={{ marginLeft: "430px" }}>
              <InputGroup size="sm">
                <Button
                  data-testid="first"
                  type="button"
                  className="button"
                  variant="outline-dark"
                  disabled={currentPage === 0 ? true : false}
                  onClick={firstPage}
                >
                  First
                </Button>
                <Button
                  data-testid="prev"
                  type="button"
                  className="button"
                  variant="outline-dark"
                  disabled={currentPage === 0 ? true : false}
                  onClick={prevPage}
                >
                  Prev
                </Button>
                <Button
                  data-testid="next"
                  type="button"
                  className="button"
                  variant="outline-dark"
                  disabled={currentPage === totalPages - 1 ? true : false}
                  onClick={nextPage}
                >
                  Next
                </Button>
                <Button
                  data-testid="last"
                  type="button"
                  className="button"
                  variant="outline-dark"
                  disabled={currentPage === totalPages - 1 ? true : false}
                  onClick={lastPage}
                >
                  Last
                </Button>
              </InputGroup>
            </div>
          </Card.Footer>
        </>
      ) : (
        <div className="no-cashbacks">No Cashbacks</div>
      )}
    </Layout>
  );
};

export default Cashback;
