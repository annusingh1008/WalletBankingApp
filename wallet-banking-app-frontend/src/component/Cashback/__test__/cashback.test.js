import React from "react";
import { fireEvent } from "@testing-library/react";
import Cashback from "../index";
import { getAllCashBacks } from "../../../actions/cashback.actions";
import { localStorageMock, renderWithRedux } from "../../../helper/testHelper";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../Layout", () =>
  jest.fn(({ children }) => <div data-testid="Layout">{children}</div>)
);

jest.mock("react-bootstrap", () => {
  const Table = jest.fn(({ children }) => (
    <table data-testid="Table">{children}</table>
  ));
  const Card = jest.fn(({ children }) => (
    <div data-testid="Card">{children}</div>
  ));
  Card.Footer = jest.fn(({ children }) => (
    <div data-testid="Footer">{children}</div>
  ));
  const InputGroup = jest.fn(({ children }) => (
    <div data-tesid="InputGroup">{children}</div>
  ));
  const Button = jest.fn((props) => (
    <button
      onClick={props.onClick}
      data-testid={`Button-${props["data-testid"]}`}
    >
      {props.children}
    </button>
  ));

  return { Table, Card, InputGroup, Button };
});

jest.mock("../../../actions/cashback.actions", () => ({
  getCashbacks: jest.fn(() => ({
    cashbackList: [
      {
        email: "abc@gmail.com",
        prev_amount: 15000,
        cashback_amount: 1000,
        current_amount: 16000,
        date: "2022-10-07T09:47:02.906+00:00",
      },
      {
        email: "abc2@gmail.com",
        prev_amount: 15000,
        cashback_amount: 1000,
        current_amount: 16000,
        date: "2022-10-07T09:47:02.906+00:00",
      },
      {
        email: "abc3@gmail.com",
        prev_amount: 15000,
        cashback_amount: 1000,
        current_amount: 16000,
        date: "2022-10-07T09:47:02.906+00:00",
      },
    ],
    totalPages: 10,
  })),
  getTotalCashbacks: jest.fn((payload) => ({
    type: "getTotalCashbacks",
    payload,
  })),
  getAllCashBacks: jest.fn((payload) => ({
    type: "getAllCashBacks",
    payload,
  })),
}));

describe("Cashback", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  Object.defineProperty(window, "localStorage", { value: localStorageMock });

  it("renders properly", () => {
    const { container } = renderWithRedux(<Cashback />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("When clicks on first button", () => {
    it("dispatches proper action", () => {
      const { queryByTestId } = renderWithRedux(<Cashback />);
      const node = queryByTestId("Button-first");
      fireEvent.click(node);
      expect(mockDispatch).toBeCalledWith(getAllCashBacks(0, "annu"));
    });
  });

  describe("When clicks on first button", () => {
    it("dispatches proper action", () => {
      const { queryByTestId } = renderWithRedux(<Cashback />);
      const node = queryByTestId("Button-prev");
      fireEvent.click(node);
      expect(mockDispatch).toBeCalledWith(getAllCashBacks(0, "annu"));
    });
  });

  describe("When clicks on first button", () => {
    it("dispatches proper action", () => {
      const { queryByTestId } = renderWithRedux(<Cashback />);
      const node = queryByTestId("Button-next");
      fireEvent.click(node);
      expect(mockDispatch).toBeCalledWith(getAllCashBacks(0, "annu"));
    });
  });

  describe("When clicks on first button", () => {
    it("dispatches proper action", () => {
      const { queryByTestId } = renderWithRedux(<Cashback />);
      const node = queryByTestId("Button-last");
      fireEvent.click(node);
      expect(mockDispatch).toBeCalledWith(getAllCashBacks(0, "annu"));
    });
  });
});
