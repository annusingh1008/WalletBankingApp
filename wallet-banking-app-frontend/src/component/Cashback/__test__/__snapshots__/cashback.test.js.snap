// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cashback renders properly 1`] = `
<div
  data-testid="Layout"
>
  <table
    data-testid="Table"
  >
    <thead
      class="text-center"
    >
      <tr>
        <th>
          #
        </th>
        <th>
          Cashback Earned
        </th>
        <th>
          Current Amount
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody
      class="text-center"
    >
      <tr>
        <td>
          1
        </td>
        <td>
          1000
        </td>
        <td>
          16000
        </td>
        <td>
          07/10/2022, 20:47:02
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          1000
        </td>
        <td>
          16000
        </td>
        <td>
          07/10/2022, 20:47:02
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          1000
        </td>
        <td>
          16000
        </td>
        <td>
          07/10/2022, 20:47:02
        </td>
      </tr>
    </tbody>
  </table>
  <div
    data-testid="Footer"
  >
    <div
      style="margin-left: 450px;"
    >
      Showing Page 
      1
       of 
      1
    </div>
    <div
      style="margin-left: 430px;"
    >
      <div
        data-tesid="InputGroup"
      >
        <button
          data-testid="Button-first"
        >
          First
        </button>
        <button
          data-testid="Button-prev"
        >
          Prev
        </button>
        <button
          data-testid="Button-next"
        >
          Next
        </button>
        <button
          data-testid="Button-last"
        >
          Last
        </button>
      </div>
    </div>
  </div>
</div>
`;
